### PR TITLE
New Feature: Insted of showing dashboard, redirect to somewhere else

### LINKED ISSUE #5297: May the dashboard be optionnal (... and redirect to an Admin)
## Feature Request

Hello, 

What are you thinking about let the Dashboard (Admin HomePage) be optionnal ?

How ? Just by adding a config option (like `default_dashboard`) wich permit to set a Default Admin like Dashboard

If you are OK, i will submit a pull request adding this feature (i want it so i will dev it anyway).

Thanks for ur amazing job :)

### PR DESCRIPTION
**Pull request by RobinDev**
I am targeting this branch, because I think it's backwards compatible.

Closes #5297

## Changelog

```markdown
### Added
- New feature: Insted of showing dashboard, it's now possible to redirect to somewhere else via a config parameter `sonata_admin.dashboard.redirect`

```

### CODE REVIEW

File: src/Action/DashboardAction.php [hunk 1]
```diff
@@ -39,20 +39,33 @@
...
+     */
+    private $redirect;
+
     public function __construct(
         array $dashboardBlocks,
         BreadcrumbsBuilderInterface $breadcrumbsBuilder,
         TemplateRegistryInterface $templateRegistry,
-        Pool $pool
+        Pool $pool,
+        string $redirect = ''
```
Comment (core23) [inline src/Action/DashboardAction.php]: We still support PHP 5.6 on the stable branch :/

Comment (RobinDev) [reply src/Action/DashboardAction.php]: What do we need to change to support PHP 5.6 ?

**Comment by greg0ire** [comment]
:thinking: why not just create a new route with the same path as the dashboard, but with higher priority? A redirect for this seems wrong…

File: src/Action/DashboardAction.php (hunk 1 above)
Comment (core23) [reply src/Action/DashboardAction.php]: This:

```suggestion
        $redirect = ''
```

**Comment by RobinDev** [comment]
@greg0ire Thank you. That's a better idea. Eg: 
```
admin_dashboard:
    path:     /admin/dashboard
    controller: Symfony\Bundle\FrameworkBundle\Controller\RedirectController::urlRedirectAction
    defaults:
        path: /admin/page/list
        permanent: true
```

**Comment by greg0ire** [comment]
Yeah, or you could even use the controller for `/admin/page/list` directly, you know? No need for 2 http requests.

**Pull request closed without merging**